src/python/build_management/revisions.py

```diff
@@ -700,3 +701,24 @@ def write_revision_to_revision_file(revision_file, revision):
 def revision_pattern_from_build_bucket_path(bucket_path):
   """Get the revision pattern from a build bucket path."""
   return '.*' + os.path.basename(bucket_path)
+
+
+@memoize.wrap(memoize.FifoOnDisk(DISK_CACHE_SIZE))
+@memoize.wrap(memoize.Memcache(60 * 60 * 24 * 30))  # 30 day TTL
+def revision_to_branched_from(uri, revision):
+  """Interrogates git code review server to find the branch-from
+  revision of a component."""
+  full_uri = "%s/+/%s?format=JSON" % (uri, revision)
+  url_content = _get_url_content(full_uri)
+  # Hatefully, gerrit returns nonsense in the first line.
```

Comment:
what does it return in the first line? if this is a bug that could be fixed in the future, is there a more robust way to detect where valid data starts?

Reply:
Yes, my comment was hateful :) I've provided a link to the gerrit API documentation where it explains why it does this.